paimon-spark/paimon-spark-common/src/main/java/org/apache/paimon/spark/SparkTypeUtils.java

```diff
@@ -285,7 +285,8 @@ public org.apache.paimon.types.DataType atomic(DataType atomic) {
                 return new CharType(((org.apache.spark.sql.types.CharType) atomic).length());
             } else if (atomic instanceof org.apache.spark.sql.types.DateType) {
                 return new DateType();
-            } else if (atomic instanceof org.apache.spark.sql.types.TimestampType) {
+            } else if (atomic instanceof org.apache.spark.sql.types.TimestampType
+                    || atomic instanceof org.apache.spark.sql.types.TimestampNTZType) {
```

Comment:
`TimestampNTZType` is introduced in spark 3.4.
Here spark 3.1 3.2 3.3 will fail

Reply:
if we define a column with type `TIMESTAMP_NTZ` in spark 3.3 (or 3.2, 3.1) ,it will throw an exception by spark directly , It will not enter the `SparkTypeUtils.SparkToPaimonTypeVisitor#atomic` method,  if we define a column with `Timestamp` in spark 3.3 ,I test it , the method could deal it correctly.